testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java

```diff
@@ -501,6 +505,87 @@ public void testOrcTableCreatedInTrino()
         onTrino().executeQuery("DROP TABLE orc_table_created_in_trino");
     }
 
+    @Test(dataProvider = "storageFormats")
+    public void testNestedFieldsWrittenByHive(String format)
+    {
+        testNestedFields(format, Engine.HIVE);
+    }
+
+    @Test(dataProvider = "storageFormats")
+    public void testNestedFieldsWrittenByTrino(String format)
+    {
+        testNestedFields(format, Engine.TRINO);
+    }
+
+    private void testNestedFields(String format, Engine writer)
+    {
+        String tableName = "test_nested_fields_written_by_" + writer.name().toLowerCase(ENGLISH);
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + tableName);
+        onTrino().executeQuery("CREATE TABLE " + tableName + " (" +
+                "  r row(a int), " +
+                "  rr row(r row(a int)), " +
+                "  ra row(a array(int)), " +
+                "  dummy varchar) WITH (format='" + format + "')");
+
+        switch (writer) {
+            case HIVE:
+                writer.queryExecutor().executeQuery("INSERT INTO " + tableName + " SELECT " +
+                        "named_struct('a', 42), " +
+                        "named_struct('r', named_struct('a', 43)), " +
+                        "named_struct('a', array(11, 22, 33)), " +
+                        "'dummy value' " +
+                        "FROM dummy");
```

Comment:
there is a `dummy` table in HIve. ILST.

Reply:
`io.trino.tests.product.hive.TestHiveStorageFormats#ensureDummyExists`